lib/moog-require.js

```diff
@@ -163,9 +163,21 @@ module.exports = function(options) {
     if (_.has(self.bundled, type)) {
       return self.bundled[type];
     }
+    // Even if the package exists in node_modules it might just be a sub-dependency
+    // due to npm flattening, which means we could be confused by an unrelated npm module
+    // with the same name as an Apostrophe module unless we verify it is a real
+    // project-level dependency
+    if (!self.validPackages) {
+      const info = JSON.parse(fs.readFileSync(`${path.dirname(self.root.filename)}/package.json`, 'utf8'));
+      self.validPackages = new Set([ ...Object.keys(info.dependencies || {}), Object.keys(info.devDependencies || {}) ]);
+    }
+    if (!self.validPackages.has(type)) {
```

Comment:
Sets: they have an API! They don't always have to be converted to arrays for my lazy brain